### PR TITLE
Added the docker compose for postgres

### PR DESCRIPTION
**Pull request by Vj-L**
Added the docker-compose file for Postgres

### CODE REVIEW

File: docker/local-metadata/docker-compose-postgres.yml [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    ports:
+      - 8080:8080
+    networks:
+      - local_app_net
+    extra_hosts:
+      - "localhost:172.16.239.10"
+      - "localhost:172.16.239.11"
+      - "localhost:172.16.239.13"
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock
```
Comment (akash-jain-10) [inline docker/local-metadata/docker-compose-postgres.yml:151]: this is not required anymore! we can get rid of this! cc - @pmbrull @ulixius9

File: docker/local-metadata/docker-compose-postgres.yml [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+      OM_AUTH_AIRFLOW_OKTA_SCOPES: ${OM_AUTH_AIRFLOW_OKTA_SCOPES:-[]}
+      # OpenMetadata Airflow Auth0 SSO Configuration
+      OM_AUTH_AIRFLOW_AUTH0_CLIENT_ID: ${OM_AUTH_AIRFLOW_AUTH0_CLIENT_ID:-""}
+      OM_AUTH_AIRFLOW_AUTH0_CLIENT_SECRET: ${OM_AUTH_AIRFLOW_AUTH0_CLIENT_SECRET:-""}
+      OM_AUTH_AIRFLOW_AUTH0_DOMAIN_URL: ${OM_AUTH_AIRFLOW_AUTH0_DOMAIN_URL:-""}
+      # OpenMetadata Airflow Custom OIDC SSO Configuration
+      OM_AUTH_AIRFLOW_CUSTOM_OIDC_CLIENT_ID: ${OM_AUTH_AIRFLOW_CUSTOM_OIDC_CLIENT_ID:-""}
+      OM_AUTH_AIRFLOW_CUSTOM_OIDC_SECRET_KEY: ${OM_AUTH_AIRFLOW_CUSTOM_OIDC_SECRET_KEY:-""}
+      OM_AUTH_AIRFLOW_CUSTOM_OIDC_TOKEN_ENDPOINT_URL: ${OM_AUTH_AIRFLOW_CUSTOM_OIDC_TOKEN_ENDPOINT_URL:-""}
+      DB_DRIVER_CLASS: ${DB_DRIVER_CLASS:-org.postgresql.Driver}
```
Comment (akash-jain-10) [inline docker/local-metadata/docker-compose-postgres.yml:102]: Add a comment above for these section to be used as Database Configurations!

File: docker/local-metadata/docker-compose-postgres.yml [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+      local_app_net:
+        ipv4_address: 172.16.239.10
+
+  elasticsearch:
+    image: docker.elastic.co/elasticsearch/elasticsearch:7.10.2
+    container_name: openmetadata_elasticsearch
+    environment:
+      - discovery.type=single-node
+      - ES_JAVA_OPTS=-Xms1024m -Xmx1024m
+
```
Comment (akash-jain-10) [inline docker/local-metadata/docker-compose-postgres.yml:42]: remove the extra spacing here!

File: docker/local-metadata/docker-compose-postgres.yml [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  postgres:
+    build:
+      context: ../../.
+      dockerfile: docker/local-metadata/Dockerfile_postgres
+    container_name: openmetadata_postgres
+    restart: always
+    depends_on:
+      - elasticsearch
+    environment:
+      POSTGRES_PASSWORD: password
```
Comment (akash-jain-10) [inline docker/local-metadata/docker-compose-postgres.yml:27]: do we want `POSTGRES_USERNAME` like environment here ? cc - @vivekratnavel

File: docker/local-metadata/docker-compose-postgres.yml [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+  ingestion-volume-dags:
+  ingestion-volume-tmp:
+services:
+  postgres:
+    build:
+      context: ../../.
+      dockerfile: docker/local-metadata/Dockerfile_postgres
+    container_name: openmetadata_postgres
+    restart: always
+    depends_on:
```
Comment (akash-jain-10) [inline docker/local-metadata/docker-compose-postgres.yml:24]: why is postgres dependent on elasticsearch here ?

**Review by akash-jain-10 — CHANGES_REQUESTED** [review]

File: docker/local-metadata/docker-compose-postgres.yml (hunk 3 above)
Comment (Vj-L) [reply docker/local-metadata/docker-compose-postgres.yml:42]: Done

File: docker/local-metadata/docker-compose-postgres.yml (hunk 2 above)
Comment (Vj-L) [reply docker/local-metadata/docker-compose-postgres.yml:102]: Done

File: docker/local-metadata/docker-compose-postgres.yml (hunk 4 above)
Comment (vivekratnavel) [reply docker/local-metadata/docker-compose-postgres.yml:27]: Yes, we need username as well in the environment

File: docker/local-metadata/docker-compose-postgres.yml [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+      OM_AUTH_AIRFLOW_AUTH0_CLIENT_SECRET: ${OM_AUTH_AIRFLOW_AUTH0_CLIENT_SECRET:-""}
+      OM_AUTH_AIRFLOW_AUTH0_DOMAIN_URL: ${OM_AUTH_AIRFLOW_AUTH0_DOMAIN_URL:-""}
+      # OpenMetadata Airflow Custom OIDC SSO Configuration
+      OM_AUTH_AIRFLOW_CUSTOM_OIDC_CLIENT_ID: ${OM_AUTH_AIRFLOW_CUSTOM_OIDC_CLIENT_ID:-""}
+      OM_AUTH_AIRFLOW_CUSTOM_OIDC_SECRET_KEY: ${OM_AUTH_AIRFLOW_CUSTOM_OIDC_SECRET_KEY:-""}
+      OM_AUTH_AIRFLOW_CUSTOM_OIDC_TOKEN_ENDPOINT_URL: ${OM_AUTH_AIRFLOW_CUSTOM_OIDC_TOKEN_ENDPOINT_URL:-""}
+      DB_DRIVER_CLASS: ${DB_DRIVER_CLASS:-org.postgresql.Driver}
+      DB_SCHEME: ${DB_SCHEME:-postgresql}
+      DB_USE_SSL: ${DB_USE_SSL:-false}
+      MYSQL_USER: ${MYSQL_USER:-openmetadata_user}
```
Comment (vivekratnavel) [inline docker/local-metadata/docker-compose-postgres.yml:105]: Why do we still need these MYSQL specific variables?

**Comment by vivekratnavel** [comment]
We need to also update the openmetadata start script to use the DB environment variables

File: docker/local-metadata/docker-compose-postgres.yml (hunk 4 above)
Comment (Vj-L) [reply docker/local-metadata/docker-compose-postgres.yml:28]: Done

File: ingestion/Dockerfile [hunk 7]
```diff
@@ -31,7 +32,13 @@ WORKDIR /ingestion
 COPY ingestion /ingestion
 
 ARG INGESTION_DEPENDENCY=all
-RUN pip install ".[${INGESTION_DEPENDENCY}]"
+RUN pip install ".[postgres,mysql,redshift]"
```
Comment (akash-jain-10) [inline ingestion/Dockerfile:35]: @ulixius9 Let's update this to be `pip install ".[${INGESTION_DEPENDENCY}]"

File: ingestion/Dockerfile [hunk 8]
```diff
@@ -31,7 +32,13 @@ WORKDIR /ingestion
...
+
+# Uninstalling psycopg2-binary and installing psycopg2 instead 
+# because the psycopg2-binary generates a architecture specific error 
+# while authrenticating connection with the airflow, psycopg2 solves this error 
+RUN pip uninstall psycopg2-binary -y
```
Comment (akash-jain-10) [inline ingestion/Dockerfile:40]: Can we remove this from `setup.py` postgres dependency itself instead ?

File: ingestion/Dockerfile [hunk 9]
```diff
@@ -31,7 +32,13 @@ WORKDIR /ingestion
...
+RUN pip install psycopg2
```
Comment (akash-jain-10) [inline ingestion/Dockerfile:41]: As per comments above, did you mean `pip uninstall` instead ?

**Review by akash-jain-10 — CHANGES_REQUESTED** [review]

File: ingestion/Dockerfile (hunk 7 above)
Comment (ulixius9) [reply ingestion/Dockerfile:35]: Opps!

File: ingestion/Dockerfile (hunk 8 above)
Comment (Vj-L) [reply ingestion/Dockerfile:40]: This have been taken care

File: docker/local-metadata/docker-compose-postgres.yml (hunk 6 above)
Comment (Vj-L) [reply docker/local-metadata/docker-compose-postgres.yml]: This has been takencare

File: ingestion/Dockerfile (hunk 9 above)
Comment (akash-jain-10) [reply ingestion/Dockerfile:41]: cc - @pmbrull ^^

File: ingestion/Dockerfile (hunk 8 above)
Comment (akash-jain-10) [reply ingestion/Dockerfile:40]: cc - @pmbrull ^^

File: docker/local-metadata/docker-compose-postgres.yml (hunk 5 above)
Comment (Vj-L) [reply docker/local-metadata/docker-compose-postgres.yml:24]: Done

File: docker/local-metadata/docker-compose-postgres.yml (hunk 1 above)
Comment (Vj-L) [reply docker/local-metadata/docker-compose-postgres.yml:160]: Done

File: ingestion/Dockerfile (hunk 8 above)
Comment (pmbrull) [reply ingestion/Dockerfile:40]: oh yeah we should do that instead, feels weird having that here

File: ingestion/Dockerfile (hunk 9 above)
Comment (pmbrull) [reply ingestion/Dockerfile:41]: let's see if we can use `psycopg2` directly on the setup

Comment (pmbrull) [reply ingestion/Dockerfile:41]: after checking with @ulixius9, agree with the idea of leaving this on the Docker not to have any unnecessary installs from users on other environments.

Thanks folks

**Comment by akash-jain-10** [comment]
@Vj-L  - Please rebase the branch from main to fix the cypress and py tests of CI. LGTM.

File: ingestion/Dockerfile [hunk 10]
```diff
@@ -1,7 +1,7 @@
 FROM python:3.9-slim as base
 ENV AIRFLOW_HOME=/airflow
 RUN apt-get update && \
-    apt-get install -y gcc libsasl2-modules libxml2 libsasl2-dev build-essential libssl-dev libffi-dev librdkafka-dev unixodbc-dev python3.9-dev openjdk-11-jre unixodbc freetds-dev freetds-bin tdsodbc libevent-dev wget openssl --no-install-recommends && \
+    apt-get install -y gcc netcat libpq-dev libxml2 postgresql postgresql-contrib libsasl2-modules libsasl2-dev build-essential libssl-dev libffi-dev librdkafka-dev unixodbc-dev python3.9-dev openjdk-11-jre unixodbc freetds-dev freetds-bin tdsodbc libevent-dev wget openssl --no-install-recommends && \
```
Comment (akash-jain-10) [inline ingestion/Dockerfile:4]: @Vj-L Let's update `docker/airflow/Dockerfile` with same apt-get install.

**Review by akash-jain-10 — CHANGES_REQUESTED** [review]

**Comment by nahuelverdugo** [comment]
@Vj-L @akash-jain-10 can we close this one? I think https://github.com/open-metadata/OpenMetadata/pull/7025 cover the same changes

**Comment by akash-jain-10** [comment]
> @Vj-L @akash-jain-10 can we close this one? I think #7025 cover the same changes

Yes!

**Pull request closed without merging**